src/main/java/sirius/db/es/ElasticQuery.java

```diff
@@ -1439,32 +1439,32 @@ protected Iterator<E> pullNextBlock() {
                 return null;
             }
 
-            JSONObject scrollResponse =
-                    scrollId == null ? pullFirstBlock() : client.continueScroll(SCROLL_TTL_SECONDS, scrollId);
-            scrollId = scrollResponse.getString(KEY_SCROLL_ID);
-            lastScroll = performScrollMonitoring(lastScroll);
-            return scrollResponse.getJSONObject(KEY_HITS)
-                                 .getJSONArray(KEY_HITS)
-                                 .stream()
-                                 .map(obj -> (E) Elastic.make(descriptor, (JSONObject) obj))
-                                 .iterator();
-        }
-
-        private JSONObject pullFirstBlock() {
+            String alias = computeEffectiveIndexName(elastic::determineReadAlias);
             String filterRouting = checkRouting(Elastic.RoutingAccessMode.READ);
-            return client.createScroll(computeEffectiveIndexName(elastic::determineReadAlias),
-                                       filterRouting,
-                                       0,
-                                       filterRouting != null ?
-                                       MAX_SCROLL_RESULTS_FOR_SINGLE_SHARD :
-                                       MAX_SCROLL_RESULTS_PER_SHARD,
-                                       SCROLL_TTL_SECONDS,
-                                       buildPayload());
+            if (pit == null) {
+                // first call to this method
+                orderAsc(ElasticEntity.ID);
+                pit = client.createPit(alias, filterRouting, "30m");
```

Comment:
Make constant of magic string 30m, and maybe 5m should be enough?

Reply:
I'd stay at 30m - the PIT is closed manually once the stream is processed, so we don't have to worry about that much, and just in case some stream processing takes a lot of time I'd rather keep the PIT open.